src/webui/api/searchcontroller.h

```diff
@@ -64,9 +65,10 @@ private slots:
 
     void checkForUpdatesFinished(const QHash<QString, PluginVersion> &updateInfo);
     void checkForUpdatesFailed(const QString &reason);
-    void searchFinished(ISession *session, int id);
-    void searchFailed(ISession *session, int id);
     int generateSearchId() const;
     QJsonObject getResults(const QList<SearchResult> &searchResults, bool isSearchActive, int totalResults) const;
     QJsonArray getPluginsInfo(const QStringList &plugins) const;
+
+    QSet<int> m_activeSearches;
+    QMap<int, std::shared_ptr<SearchHandler>> m_searchHandlers;
```

Comment:
Is it possible to use `std::unique_ptr` instead? I don't think you need to pass around the object.

Reply:
> Is it possible to use `std::unique_ptr` instead? I don't think you need to pass around the object.

Shared pointer (actually QSharedPointer) was used earlier. I'd rather not touch it here. Unlike other proposed changes, this may not be so trivial.

Reply:
All other comments are addressed.

Reply:
seems qt containers are using copy semantics and therefore is not compatible with std::unique_ptr which requires move semantics.
https://stackoverflow.com/questions/70904952/how-to-make-unique-ptr-work-with-qts-qhash-and-qstring